rfcs/proposed/simplified_numa_support/README.md

```diff
@@ -0,0 +1,179 @@
+# Simplified NUMA support in oneTBB
+
+## Introduction
+
+In Non-Uniform Memory Access (NUMA) systems, the cost of memory accesses depends on the
+*nearness* of the processor to the memory resource on which the accessed data resides. 
+While oneTBB has core support that enables developers to tune for Non-Uniform Memory 
+Access (NUMA) systems, we believe this support can be simplified and improved to provide 
+an improved user experience.  
+
+This early proposal recommends addressing for areas for improvement:
+
+1. improved reliability of HWLOC-dependent topology and pinning support in,
+2. addition of a NUMA-aware allocation,
+3. simplified approaches to associate task distribution with data placement and 
+4. where possible, improved out-of-the-box performance for high-level oneTBB features.
+
+We expect that this draft proposal may be broken into smaller proposals based on feedback 
+and prioritization of the suggested features.
+
+The features for NUMA tuning already available in the oneTBB 1.3 specification include:
+
+- Functions in the `tbb::info` namespace **[info_namespace]** 
+  - `std::vector<numa_node_id> numa_nodes()`
+  - `int default_concurrency(numa_node_id id = oneapi::tbb::task_arena::automatic)`
+- `tbb::task_arena::constraints` in **[scheduler.task_arena]**
+
+Below is the example that demonstrates the use of these APIs to pin threads to different 
+arenas to each of the NUMA nodes available on a system, submit work across those `task_arena` 
+objects and into associated `task_group`` objects, and then wait for work again using both 
+the `task_arena` and `task_group` objects.
```

Comment:
I think the code can be made simpler with `std::thread` per domain, instead of relying only TBB. On the one hand, it also signals that TBB lacks high-level NUMA APIs. On the other hand, TBB, and task arenas specifically, were designed to work well with application level threads where it makes sense. I think it is much better to assume/suggest each NUMA aware arena to be used by a special application thread than to add extra levels of complication with task groups.

Reply:
This pattern of `task_arenas` and `task_groups` is what we show in our documentation: [for example here](https://oneapi-src.github.io/oneTBB/main/tbb_userguide/Guiding_Task_Scheduler_Execution.html). And, probably as a consequence, a pattern we see in applications that use NUMA constraints.